.github/actions/create-project-issue-v1/src/run.test.ts

```diff
@@ -0,0 +1,308 @@
+          })
+        }
+      }
+    })
+  }
+
+  beforeEach(() => {
+    setFailed = sinon.spy()
+    setOutput = sinon.stub()
+    info = sinon.spy()
+    getInput = sinon.stub()
+    getExecOutput = sinon.stub(exec, 'getExecOutput')
+  })
+
+  afterEach(sinon.restore)
+
+  const createAndMoveIssueStub = async () => {
+    //`getProjectBoardID` stub
+    getExecOutput.onFirstCall().resolves({
+      stdout: JSON.stringify(MOCK_PROJECT_BOARD_ID),
+      exitCode: 0
+    })
+
+    //`addIssueToBoard` stub
+    getExecOutput.onSecondCall().resolves({
+      stdout: JSON.stringify(MOCK_ISSUE_ADDED),
+      exitCode: 0
+    })
+
+    //`getProjectFieldList` stub
+    getExecOutput.onThirdCall().resolves({
+      stdout: JSON.stringify(MOCK_FIELD_LIST),
+      exitCode: 0
+    })
+
+    //`moveIssueToColumn` stub
+    getExecOutput.onCall(3).resolves({
+      stdout: JSON.stringify(MOCK_ISSUE_MOVED),
+      exitCode: 0
+    })
+  }
+
+  describe('when the `github_token` input is not provided', () => {
+    it('throws an error', async () => {
+      getInput.withArgs('github_token', { required: true }).throws({
+        message: 'Input required and not supplied: github_token'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: github_token')
+      )
+    })
+  })
+
+  describe('when the `title` input is not provided', () => {
+    it('throws an error', async () => {
+      getInput.withArgs('title', { required: true }).throws({
+        message: 'Input required and not supplied: title'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: title')
+      )
+    })
+  })
+
+  describe('when the `body` input is not provided', () => {
+    it('throws an error', async () => {
+      getInput.withArgs('body', { required: true }).throws({
+        message: 'Input required and not supplied: body'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: body')
+      )
+    })
+  })
+
+  describe('when the `project_number` input is not a number', () => {
+    it('throws an error', async () => {
+      getInput.withArgs('github_token', { required: true }).returns('token')
+      getInput.withArgs('title', { required: true }).returns('title')
+      getInput.withArgs('body', { required: true }).returns('body')
+      getInput.withArgs('repository').returns('dequelabs/repo')
```

Comment:
These tests aren't very robust as they just ensure you got to the end but not that any of the provided inputs got used as expected (due to mocking). Instead I would look at the specific `getExecOutput` calls and make sure the expected inputs got used for the call. And also would check all the parts individually.

So for example:

```js
it('Looks at the project board using the project board id')
  getInput.withArgs('project_number').returns('70')

  assert.isTrue(getExecOutput.calledWith('gh project view 70 --owner dequelabs --format json'))
```